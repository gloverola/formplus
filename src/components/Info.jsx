import { device } from "~/styles/breakpoints";
import React from "react";
import styled from "styled-components";
import { ReactComponent as InfoIcon } from "~/assets/icons/info-icon.svg";

/**
 * * Info component
 * @returns Info alert
 */
const Info = () => {
  return (
    <InfoView>
      <InfoIcon />{" "}
      <p>
        Tada! Get started with a free template. Canâ€™t find what you are looking
        for? Search from the 1000+ available templates
      </p>
    </InfoView>
  );
};

export default Info;

const InfoView = styled.div`
  width: 100%;
  padding: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors?.amber};
  border-radius: 2px;
  margin: 60px 0;

  @media ${device.phone} {
    width: 100%;
    padding: 10px;
    justify-content: flex-start;
  }

  p {
    margin-left: 5px;
    font-size: 0.875rem;
    color: ${({ theme }) => theme.colors?.neutral_8};

    @media ${device.phone} {
      width: 90%;
    }
  }
`;
