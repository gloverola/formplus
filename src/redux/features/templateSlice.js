import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  templates: [],
  currentPage: 1,
  itemOffSet: 0,
  endOffSet: 15,
  totalTemplates: 0,
  currentItems: [],
  pageCount: 1,
  loading: false,
  category: "All",
  order: "Default",
  date: "Default",
};

const templateSlice = createSlice({
  name: "template",
  initialState,
  reducers: {
    setTemplates: (state, { payload }) => {
      state.templates = payload;
    },
    setCurrentPage: (state, { payload }) => {
      state.currentPage = payload;
    },
    setItemOffSet: (state, { payload }) => {
      state.itemOffSet = payload;
    },
    setEndOffset: (state, { payload }) => {
      state.endOffSet = payload;
    },
    setTotalTemplates: (state, { payload }) => {
      state.totalTemplates = payload;
    },
    setCurrentItems: (state, { payload }) => {
      state.currentItems = payload;
    },
    setPageCount: (state, { payload }) => {
      state.pageCount = payload;
    },
    setLoading: (state, { payload }) => {
      state.loading = payload;
    },
    setCategory: (state, { payload }) => {
      state.category = payload;
    },
    setOrder: (state, { payload }) => {
      state.order = payload;
    },
    setDate: (state, { payload }) => {
      state.date = payload;
    },

    resetAppState: () => {},
  },
});

export const {
  setTemplates,
  resetAppState,
  setCurrentPage,
  setCurrentItems,
  setItemOffSet,
  setEndOffset,
  setTotalTemplates,
  setPageCount,
  setLoading,
  setCategory,
  setOrder,
  setDate,
} = templateSlice?.actions;

export default templateSlice?.reducer;

export const templateSelector = (state) => state?.template;
