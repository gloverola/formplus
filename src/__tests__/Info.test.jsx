import { Provider } from "react-redux";
import ReactDOM from "react-dom";
import { describe, it, expect, beforeEach } from "vitest";
import templateSlice from "~/redux/features/templateSlice";
import configureStore from "redux-mock-store";
import TestRenderer from "react-test-renderer";
import Info from "~/components/Info";

// Test Header Component renders properly
let component;
describe("Info component", () => {
  const mockStore = configureStore([]);
  beforeEach(() => {
    let div = document.createElement("div");
    const store = mockStore({
      template: templateSlice,
    });
    component = ReactDOM.render(
      <Provider store={store}>
        <Info />
      </Provider>,
      div
    );
  });

  it("should render properly with given state from Redux store", () => {
    const tree = TestRenderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
