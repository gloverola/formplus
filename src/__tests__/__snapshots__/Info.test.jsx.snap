// Vitest Snapshot v1

exports[`Info component > should render properly with given state from Redux store 1`] = `null`;
